### PR TITLE
Allow changing nickname in any room

### PR DESCRIPTION
**Pull request by vranki**
Related to #1216 - the most important command is to change nick. This allows doing so from any room.

### CODE REVIEW

File: src/bridge/IrcBridge.ts [hunk 1]
```diff
@@ -1202,6 +1202,18 @@ export class IrcBridge {
             "Sending IRC message in %s as %s (connected=%s)",
```
Comment (Half-Shot) [inline src/bridge/IrcBridge.ts:1202]: I worry that this is no longer quite true.

File: src/bridge/IrcBridge.ts [hunk 2]
```diff
@@ -1202,6 +1202,18 @@ export class IrcBridge {
...
             ircRoom.channel, bridgedClient.nick, Boolean(bridgedClient.status === BridgedClientStatus.CONNECTED)
         );
+        if (action.text.startsWith("!irc nick")) {
```
Comment (Half-Shot) [inline src/bridge/IrcBridge.ts:1205]:
```suggestion
        if (action.text.startsWith("!irc nick ")) {
```

to check formatting.

File: src/bridge/IrcBridge.ts [hunk 3]
```diff
@@ -1202,6 +1202,18 @@ export class IrcBridge {
...
+            const newNick = action.text.substr(10)
```
Comment (Half-Shot) [inline src/bridge/IrcBridge.ts:1206]: This is easier to read

```suggestion
            const newNick = action.text.substr("!irc nick ".length)
```

File: src/bridge/IrcBridge.ts [hunk 4]
```diff
@@ -1202,6 +1202,18 @@ export class IrcBridge {
...
+            log.info("Matrix user wants to change nick from %s to %s", bridgedClient.nick, newNick)
+            try {
+               return bridgedClient.changeNick(newNick, true)
```
Comment (Half-Shot) [inline src/bridge/IrcBridge.ts:1209]: I wonder if we should await this one, and give the user some feedback if they couldn't change their nick?

**Review by Half-Shot — COMMENTED** [review]
Some thoughts, looks handy though.

File: src/bridge/IrcBridge.ts (hunk 4 above)
Comment (vranki) [reply src/bridge/IrcBridge.ts]: I suck at JavaScript and failed to do this. Added a todo comment.

**Comment by vranki** [comment]
I did the requested changes. Someone more skilled could add the result message, but at least the nick change functionality works now.

**Review by vranki — COMMENTED** [review]
Nice, this is better architecture allowing more commands to be implemented in future cleanly.

**Comment by Half-Shot** [comment]
(Merging as we've both reviewed it)

**Pull request merged**